mlos_bench/mlos_bench/services/remote/azure/azure_fileshare.py

```diff
@@ -52,23 +53,30 @@ def __init__(
             parent,
             self.merge_methods(methods, [self.upload, self.download]),
         )
-
         check_required_params(
             self.config,
             {
                 "storageAccountName",
                 "storageFileShareName",
-                "storageAccountKey",
             },
         )
-
-        self._share_client = ShareClient.from_share_url(
-            AzureFileShareService._SHARE_URL.format(
-                account_name=self.config["storageAccountName"],
-                fs_name=self.config["storageFileShareName"],
-            ),
-            credential=self.config["storageAccountKey"],
-        )
+        self._share_client: Optional[ShareClient] = None
+
+    def _get_share_client(self) -> ShareClient:
+        """Get the Azure file share client object."""
+        if self._share_client is None:
+            assert self._parent is not None and isinstance(
+                self._parent, SupportsAuth
+            ), "Authorization service not provided. Include service-auth.jsonc?"
```

Comment:
I think that check should be in `__init__` for earlier feedback on bad configs.

Reply:
I wish I could move all that functionality to `__init__()` where it actually belongs! Unfortunately, that breaks our unit tests that instantiate each Service class with test configurations to validate the JSONs. `_get_share_client()` is just a workaround for this, just like a bunch of other methods with delayed initialization in other Services, Environments, etc.

Reply:
Proposed fix to that:
https://github.com/motus/MLOS/pull/14

Reply:
(we already do that in one or two other places)